### PR TITLE
Add infrastructure

### PR DESCRIPTION
**Pull request by inbarbarkai**
* Basic web application code.
* Ingress configuration for the application and API.
* OpenFAAS initialization scripts.
* Docker compose configuration for all basic services.
* Sample OpenFAAS function.

### CODE REVIEW

File: local/install-environment.ps1 [hunk 1]
```diff
@@ -0,0 +1,67 @@
+param(
```
Comment (yarons) [inline local/install-environment.ps1:1]: Are you sure about this implementation?
Maybe we should go with [`arkade`](https://github.com/alexellis/arkade/releases) and create some wrapping around it?

File: local/k8s/k8s-cluster.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+containerdConfigPatches:
+- |-
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${reg_port}"]
+    endpoint = ["http://host.docker.internal:5000"]
```
Comment (yarons) [inline local/k8s/k8s-cluster.yml:6]: I've had some issues with this address, from what I understood the preferred method is to use `.svc` subdomain across namespaces (`http://host.svc.docker.internal:5000`) and one more thing, after your modifications, shouldn't it be `https`?

**Review by yarons — COMMENTED** [review]
Looking good, I will need to destroy and re-implement my `kind` deployment and test the whole implementation.

File: app/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM node:21-alpine3.17 as builder
+
+ADD . /workdir
+
+WORKDIR /workdir
+
+RUN yarn install && yarn run build
+
+
+FROM nginx:1.24.0-alpine
```
[the remark this answers is not in the document]
Comment (inbarbarkai) [reply app/Dockerfile:10]: I'm not fixed on alpine, feel free to replace to whatever you want

File: app/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
+    "compilerOptions": {
+        "target": "es5",
```
[the remark this answers is not in the document]
Comment (inbarbarkai) [reply app/tsconfig.json:3]: Copy/Paste mistake

File: local/k8s/k8s-cluster.yml (hunk 2 above)
Comment (inbarbarkai) [reply local/k8s/k8s-cluster.yml:6]: I think I can remove it.
I need to test.

File: local/install-environment.ps1 (hunk 1 above)
Comment (inbarbarkai) [reply local/install-environment.ps1:1]: I will take a look at it.

File: local/k8s/k8s-cluster.yml (hunk 2 above)
Comment (yarons) [reply local/k8s/k8s-cluster.yml:6]: God blast you kapara :)

File: local/install-environment.ps1 (hunk 1 above)
Comment (inbarbarkai) [reply local/install-environment.ps1:1]: It fails to install due to reliance on Linux/Unix commands:
https://github.com/alexellis/arkade/issues/977

File: local/k8s/k8s-cluster.yml (hunk 2 above)
Comment (inbarbarkai) [reply local/k8s/k8s-cluster.yml:6]: Removed

File: app/tsconfig.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (inbarbarkai) [reply app/tsconfig.json]: Fixed

**Pull request merged**